test/cases/adapters/postgresql/postgresql_adapter_test.rb

```diff
@@ -63,23 +63,34 @@ def test_using_follower_reads_connects_properly
         assert conn_config[:use_follower_reads_for_type_introspection]
       end
 
-      def test_only_reload_type_map_once_for_every_unrecognized_type
-        reset_connection
-        connection = ActiveRecord::Base.connection
+      # OVERRIDE: CockroachDB adds parentheses around the WHERE clause's content.
+      def test_partial_index_on_column_named_like_keyword
+        with_example_table('id serial primary key, number integer, "primary" boolean') do
+          @connection.add_index "ex", "id", name: "partial", where: "primary" # "primary" is a keyword
+          index = @connection.indexes("ex").find { |idx| idx.name == "partial" }
+          assert_equal '("primary")', index.where
+        end
+      end
 
-        silence_stream($stdout) do
-          assert_queries 2, ignore_none: true do
-            connection.select_all "select 'pg_catalog.pg_class'::regclass"
-          end
-          assert_queries 1, ignore_none: true do
-            connection.select_all "select 'pg_catalog.pg_class'::regclass"
-          end
-          assert_queries 2, ignore_none: true do
-            connection.select_all "SELECT NULL::anyarray"
+      def test_invalid_index
+        with_example_table do
+          @connection.exec_query("INSERT INTO ex (number) VALUES (1), (1)")
+          error = assert_raises(ActiveRecord::RecordNotUnique) do
+            @connection.add_index(:ex, :number, unique: true, algorithm: :concurrently, name: :invalid_index)
           end
+          assert_match(/duplicate key value violates unique constraint/, error.message)
+          assert_equal @connection.pool, error.connection_pool
+
+          # TODO: It seems like in CRDB this tests won't create the index at all.
+          #   It is created in PG with `valid: false` set. Is this correct?
```

Comment:
So here there is a behaviour difference between CRDB and PG:

PG creates the index anyway, but invalid. CRDB doesn't create it. Should I dig into it to see if this is a CRDB bug, or something we want ?

Reply:
this is an area where CRDB diverges from PG. see: https://github.com/cockroachdb/cockroach/issues/65929

Reply:
So I should modify the test to look for a constraint then ? And add a comment referencing this issue maybe

Reply:
FYI @rafiss I've looked at what is generated and I see neither and index nor a constraint

Reply:
ah i see; i read the test more closely now. yes, that's another difference in CRDB. if you try to add the unique index but it fails, then neither the index nor constraint will exist.